src/main/java/it/pagopa/pm/gateway/service/VposDeleteService.java

```diff
@@ -0,0 +1,159 @@
+package it.pagopa.pm.gateway.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.pagopa.pm.gateway.client.vpos.HttpClient;
+import it.pagopa.pm.gateway.client.vpos.HttpClientResponse;
+import it.pagopa.pm.gateway.dto.creditcard.StepZeroRequest;
+import it.pagopa.pm.gateway.dto.enums.RefundOutcome;
+import it.pagopa.pm.gateway.dto.vpos.AuthResponse;
+import it.pagopa.pm.gateway.dto.vpos.VposDeleteResponse;
+import it.pagopa.pm.gateway.dto.vpos.VposOrderStatusResponse;
+import it.pagopa.pm.gateway.entity.PaymentRequestEntity;
+import it.pagopa.pm.gateway.repository.PaymentRequestRepository;
+import it.pagopa.pm.gateway.utils.VPosRequestUtils;
+import it.pagopa.pm.gateway.utils.VPosResponseUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.BooleanUtils;
+import org.apache.http.entity.ContentType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static it.pagopa.pm.gateway.constant.Messages.*;
+import static it.pagopa.pm.gateway.constant.VposConstant.RESULT_CODE_AUTHORIZED;
+import static it.pagopa.pm.gateway.dto.enums.PaymentRequestStatusEnum.CANCELLED;
+import static it.pagopa.pm.gateway.dto.enums.RefundOutcome.KO;
+import static it.pagopa.pm.gateway.dto.enums.RefundOutcome.OK;
+
+@Service
+@Slf4j
+public class VposDeleteService {
+
+    @Value("${vpos.requestUrl}")
+    private String vposUrl;
+
+    @Autowired
+    private PaymentRequestRepository paymentRequestRepository;
+
+    @Autowired
+    private VPosRequestUtils vPosRequestUtils;
+
+    @Autowired
+    private VPosResponseUtils vPosResponseUtils;
+
+    @Autowired
+    private HttpClient httpClient;
+
+    @Autowired
+    private ObjectMapper objectMapper;
```

Comment:
Prefer constructor injection to field injection as it makes testing easier to setup

Reply:
https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/199/commits/d7dbba75be5440c060afdc231fde7caaae4ef892